hibernate-core/src/main/java/org/hibernate/sql/Template.java

```diff
@@ -205,6 +206,40 @@ else if ( quotedIdentifier && ( dialect.closeQuote()==token.charAt(0) ) ) {
 				continue;
 			}
 
+			// Special processing for type casting
+			if ( inCast && "as".equals( lcToken ) ) {
+				result.append( " as " );
+				//to manage cases like cast(field as numeric(10,4))
+				int openBrackets = 0;
+
+				hasMore = tokens.hasMoreTokens();
+				nextToken = hasMore ? tokens.nextToken() : null;
+
+				while ( hasMore && inCast ) {
+					if ( ")".equals( nextToken ) ) {
```

Comment:
@gbadner,

 I think to close this PR rejecting it in favour of https://github.com/hibernate/hibernate-orm/pull/1011
what do you think?